src/app/core/services/courses.service.ts

```diff
@@ -43,47 +47,64 @@ const coursesList = [{
 
 @Injectable()
 export class CoursesService {
+  private isPopupDisplayed;
+  private deleteCourseId;
+  private courses: BehaviorSubject<ICourse[]> = new BehaviorSubject([]);
+
+  constructor () {
+    this.isPopupDisplayed = false;
+  }
+
+  public get coursesStream (): Observable<ICourse[]> {
+    return this.courses.delay(2000);
+  }
+
   public getCourses () {
-    return coursesList;
+    this.courses.next(coursesList);
   }
 
-  public createNewCourse (course) {
+  public createNewCourse (course: ICourse) {
     coursesList.push(course);
+    this.courses.next(coursesList);
   }
 
   public getCourse(id: number) {
-    return new Promise((resolve, reject) => {
-      coursesList.forEach((item) => {
-        if (item.id === id) {
-          resolve(item);
-        }
-      });
+    const selectedElement = coursesList.find((data) => {
+      return data.id === id;
     });
+    this.courses.next([selectedElement]);
   }
 
-  public  deleteCourse (courseId) {
-    coursesList.filter((item) => {
-      if (item.id === courseId) {
-        const index = coursesList.indexOf(item);
-        if (index >= 0) {
-          coursesList.splice( index, 1 );
-        }
-      }
-    });
+  public deleteCourse (courseId): boolean {
+    this.isPopupDisplayed = true;
+    this.deleteCourseId = courseId;
+    return true;
   }
 
-  public editCourse (course) {
-    coursesList.filter((item) => {
-      if (item.id === +course.id) {
-        const index = coursesList.indexOf(item);
-        if (index >= 0) {
-          coursesList[index] = course;
-        }
-      }
-    });
+  public confirmPopupStatus (): boolean {
+    return !!this.isPopupDisplayed;
+  }
+
+  public confirmDelete (): boolean {
+    _.pullAllBy(coursesList, [{ id: this.deleteCourseId }], 'id');
+    this.isPopupDisplayed = false;
+    return false;
+  }
+
+  public rejectDelete (): boolean {
```

Comment:
teraz zauważyłem, że anulowanie usuwania nie działa: http://g.recordit.co/FezTRykR3z.gif

a tak btw. teraz zauważyłem, że masz w serwisie metody związane z popupem (`confirmPopupStatus `, `rejectDelete` itp. ), raczej nie powinno tak być, bo serwis powinien być niezależny od logiki UI, to czy ma wyskoczyć popup z pytanie czy usunąć kurs powinno być obsłużone w komponencie